aws_ros1_common/src/sdk_utils/ros1_node_parameter_reader.cpp

```diff
@@ -64,7 +64,7 @@ AwsError Ros1NodeParameterReader::ReadParam(const ParameterPath & param_path, Aw
   std::string value;
   AwsError result = ReadParam(param_path, value);
   if (result == AWS_ERR_OK) {
-    out = Aws::String(value.c_str());
+    out = Aws::String(value.c_str()); // NOLINT(readability-redundant-string-cstr)
```

Comment:
Is this a bug in clang-tidy?

Reply:
Sort of. It didn't take into consideration that although `decltype(value)` and `Aws::String` are both std::strings under the hood, they're not fully compatible because they use different allocator types. This annoyance has also been a long gripe about the C++ standard library in the C++ community.